### PR TITLE
Drop py38

### LINKED ISSUE #499: Drop python 3.8 support
Hi @CasperWA

It would be great to have #484 and #482 merged.

Currently they're failing because both aiida-core v2.4.0 and optimade-python-tools 0.25 are not compatible with 3.8 any more. 

Any pointers on best way to do this? I see that the readme table will need a bit of an overhaul.

### PR DESCRIPTION
**Pull request by eimrek**
fixes #499

### CODE REVIEW

**Comment by eimrek** [comment]
@CasperWA I also added 3.11 to the CI tests. i guess if they pass, I can add that to the list of supported python versions as well. Then the versions match with aiida-core.

**Comment by eimrek** [comment]
@CasperWA regarding the readme compatilibty matrix, we still keep versions 1.0-1.2 compatible with AiiDA 2.0-3.0, right? It should be the case, as long as you use the don't use python 3.8.

File: README.md
Comment (CasperWA) [inline README.md]: I suppose the AiiDA range here should be updated?

**Review by CasperWA — CHANGES_REQUESTED** [review]
Looks good to me - just need to update the AiiDA version range in the README for the new AiiDA-OPTIMADE version range.

Did you run `pyupgrade --py39-plus` on all the files or where do the other changes come from (related to `open()` and such)?

**Comment by CasperWA** [comment]
> @CasperWA regarding the readme compatilibty matrix, we still keep versions 1.0-1.2 compatible with AiiDA 2.0-3.0, right? It should be the case, as long as you use the don't use python 3.8.

Hmm. I'd only keep it compatible with the versions of AiiDA that support Python 3.8, I suppose?

File: README.md
Comment (CasperWA) [reply README.md]: Ah - maybe it should indeed be changed in the line below as I also suggest [here](https://github.com/aiidateam/aiida-optimade/pull/500#issuecomment-1794529906).

**Comment by eimrek** [comment]
thanks @CasperWA 

> Did you run pyupgrade --py39-plus on all the files or where do the other changes come from (related to open() and such)?

I ran `pyupgrade --py37-plus` before (that changed open()) but now also `pyupgrade --py39-plus`, which introduced many changes to the typing library imports.

Regarding the compatilibity matrix, i updated according to the following:
* aiida >=2.0.0, <2.4.0 is compatible with py3.8, so i used that range for 1.0-1.2
* aiida >=2.2.0 is compatible with py 3.11, so i used that for the new version that will also be compatible with py3.11.

Feel free to suggest alternatives.

**Review by CasperWA — APPROVED** [review]
This looks good to me.

After merging this, the currently failing dependabot PRs should be updated and hopefully they will succeed - then they should be merged and v1.2.0 should be released (probably after running CI workflows to check for any new dependency updates).

Note, the dependabot PRs should auto-update, if not, one should wait until the CI workflows running after merging this PR finish and then the PRs should be updated manually.

**Comment by eimrek** [comment]
thanks @CasperWA the PRs didnt seem to get updated automatically. I merged develop into the aiida update PR #484. Is this what you meant? If that's fine i'll do the same for #482 as well.

**Comment by CasperWA** [comment]
So I'll just finish up with merging the latest dependency updates, and then I'll release v1.2.0. Please then check everything you need is solved by v1.2.0 concerning the immediate issue(s) and subsequent requirement for dropping Python 3.8 support :)

**Comment by eimrek** [comment]
thanks @CasperWA! Everything works great!

**Comment by CasperWA** [comment]
> thanks @CasperWA! Everything works great!

Thank you! 😄

**Pull request merged**
